src/sentry/api/endpoints/project_group_index.py

```diff
@@ -277,14 +277,10 @@ def get(self, request, project):
             if len(query) == 32:
                 # check to see if we've got an event ID
                 try:
-                    mapping = EventMapping.objects.get(
-                        project_id=project.id,
-                        event_id=query,
-                    )
-                except EventMapping.DoesNotExist:
+                    matching_group = Group.objects.from_event_id(project, query)
+                except Group.DoesNotExist:
                     pass
                 else:
-                    matching_group = Group.objects.get(id=mapping.group_id)
                     try:
                         matching_event = Event.objects.get(
```

Comment:
It seems weird to have to make this query again, even though we've already made one very similar as part of `from_event_id` at this point but didn't actually return the full `Event` object. I wonder if it'd make sense to change the return type of `from_event_id` to `(group, event or None)`? I guess this would make the function marginally slower since it'd have to return all of the event columns (instead of just the group ID), as well as instantiate a new `Event` object if one exists.

Reply:
Yeah, I thought the same thing, but in any use case for reads, this is effectively exactly what the old behavior was. And the usages are so low for read path, it doesn't outweigh imo making the function return value more complex and convoluted.